microbench/src/main/java/io/netty/microbench/concurrent/BurstCostExecutorsBenchmark.java

```diff
@@ -0,0 +1,166 @@
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.microbench.concurrent;
+
+import io.netty.channel.epoll.Epoll;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import io.netty.util.concurrent.DefaultEventExecutor;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Threads;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public class BurstCostExecutorsBenchmark extends AbstractMicrobenchmark {
+
+    enum ExecutorType {
+        defaultEventExecutor,
+        juc,
+        nioEventLoop,
+        epollEventLoop
+    }
+
+    @Param({ "1", "10" })
+    private int burstLength;
+    @Param({ "epollEventLoop", "nioEventLoop", "defaultEventExecutor", "juc" })
+    private String executorType;
+
+    private ExecutorService executor;
+    private ExecutorService executorToShutdown;
+
+    @Setup
+    public void setup() {
+        ExecutorType type = ExecutorType.valueOf(executorType);
+        switch (type) {
+        case defaultEventExecutor:
+            executor = new DefaultEventExecutor();
+            executorToShutdown = executor;
+            break;
+        case juc:
+            executor = Executors.newSingleThreadScheduledExecutor();
+            executorToShutdown = executor;
+            break;
+        case nioEventLoop:
+            NioEventLoopGroup nioEventLoopGroup = new NioEventLoopGroup(1);
+            nioEventLoopGroup.setIoRatio(1);
+            executor = nioEventLoopGroup.next();
+            executorToShutdown = nioEventLoopGroup;
+            break;
+        case epollEventLoop:
+            Epoll.ensureAvailability();
+            EpollEventLoopGroup epollEventLoopGroup = new EpollEventLoopGroup(1);
+            epollEventLoopGroup.setIoRatio(1);
+            executor = epollEventLoopGroup.next();
+            executorToShutdown = epollEventLoopGroup;
+            break;
```

Comment:
@franz1981 do you also want to include Kqueue ?

Reply:
I have tried to not get any `IllegalStateException` by using 2 separate profile (linux/mac) that would use 2 different bench classes using the right epoll/kqueue even loop, but I'm having some conflicts on the pom. Do you know any other way to address it?
I can live with IllegalStateException while running the bench on my linux machine, but I would prefer to make it clean :P

Reply:
hmm as long as you not include these in the `@Params` it should be fine, no ?

Reply:
Yep, but that means that you have to always configure the bench params while running it and you can't just throw a bench as it is: I will just add kqueue without worrying about it so much.
If anyone will complain I can always change it in another form :+1: 
I will do a run for it too, because I'm curious: in the next weeks I was planning to take a look on how to improve some of them. There is anything different on Netty 5 re event loop impls?

Reply:
Yes, much to my dismay ;) https://github.com/netty/netty/pull/8587

Reply:
@johnou it seems a big change, but I do understand the concerns of @normanmaurer as well :O

Reply:
@normanmaurer @johnou 
I have notice that I was missing another thing in the bench: a baseline.
Given that I want to use this bench to check improvements/regressions I don't want to spend time (and I think is the same of others) trying to improve things without knowing what is the best that could be achieved. That's why I have put a naive implementation of single consumer executor that won't sleep and that would likely to have the best ever burst cost.
Re the conflicts on the pom, what can I do? I wanted to do with kqueue the same of epoll, but seems not possible...

Reply:
@franz1981 @johnou while #8587 does touch things related to EventLoops etc it not really change anything that should affect the benchmark here. Its more about correctness / contract / and safety.

Reply:
@franz1981 I think for your merge conflict you should be good after rebase on origin/4.1